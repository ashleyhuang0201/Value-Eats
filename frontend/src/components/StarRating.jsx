import React from "react";

// import StarBorderRoundedIcon from '@material-ui/icons/StarBorderRounded';
// import StarHalfRoundedIcon from '@material-ui/icons/StarHalfRounded';
// import StarRoundedIcon from '@material-ui/icons/StarRounded';

// import { Box, IconButton } from '@material-ui/core';

import { Rating } from "@material-ui/lab";

export default function StarRating ({ rating, isEditable = false, setRating }) {
  return (
    <Rating value={rating}
    precision={0.5}
    onChange={(event, newValue) => {
      setRating(newValue);
    }}
    readOnly={!isEditable}
    >
    </Rating>
  );
}
