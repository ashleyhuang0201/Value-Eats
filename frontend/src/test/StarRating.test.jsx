import { render, shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import StarRating from "../components/StarRating";

describe("StarRating", () => {
  const noop = () => {};

  it("can work in the base case", () => {
    const rating = shallow(<StarRating rating={5} setRating={noop}/>);
    expect(rating.exists()).toBeTruthy();
    expect(rating.find("span")).toBeTruthy();
  });

  it("can handle clicks correctly when editable", () => {
    const onClick = jest.fn();
    const rating = shallow(<StarRating rating={5} isEditable={true} setRating={onClick}/>);
    // console.log(rating.debug({ verbose: true }));
    rating.simulate("change");
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it("can handle isEditable true correctly", () => {
    const rating = render(<StarRating rating={5} isEditable={false} setRating={noop}/>);
    expect(rating.hasClass("MuiRating-readOnly")).toBeTruthy();
  });

  it("can handle isEditable false correctly", () => {
    const rating = render(<StarRating rating={5} isEditable={true} setRating={noop}/>);
    // expect(rating.hasClass("MuiRating-readOnly")).toBeTruthy();
    expect(rating.hasClass("MuiRating-readOnly")).toBeFalsy();
  });

  /* Snapshot tests - creates snapshots for new tests, otherwise will check the existing tests. To update existing tests,
  follow the prompts on the screen after running test. Done to avoid regression. */

  it("renders with minimal props", () => {
    const rating = renderer.create(<StarRating rating={5} setRating={noop}/>);
    expect(rating).toMatchSnapshot();
  });

  it("renders with double star rating", () => {
    const rating = renderer.create(<StarRating rating={2.5} setRating={noop}/>);
    expect(rating).toMatchSnapshot();
  });

  it("renders correctly when editable", () => {
    const rating = renderer.create(<StarRating rating={5} isEditable={true} setRating={noop}/>);
    expect(rating).toMatchSnapshot();
  });
});
