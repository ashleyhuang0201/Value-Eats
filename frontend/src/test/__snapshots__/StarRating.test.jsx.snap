// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRating renders correctly when editable 1`] = `
<span
  aria-label={null}
  className="MuiRating-root"
  onMouseLeave={[Function]}
  onMouseMove={[Function]}
  role={null}
>
  <span
    className="MuiRating-decimal"
  >
    <label
      className="MuiRating-label"
      htmlFor="undefined-0-5"
      style={
        Object {
          "overflow": "hidden",
          "position": "absolute",
          "width": "0%",
          "zIndex": 1,
        }
      }
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
      <span
        className="MuiRating-visuallyhidden"
      >
        0.5 Stars
      </span>
    </label>
    <input
      checked={false}
      className="MuiRating-visuallyhidden"
      disabled={false}
      id="undefined-0-5"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value={0.5}
    />
    <label
      className="MuiRating-label"
      htmlFor="undefined-1"
      style={Object {}}
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
      <span
        className="MuiRating-visuallyhidden"
      >
        1 Star
      </span>
    </label>
    <input
      checked={false}
      className="MuiRating-visuallyhidden"
      disabled={false}
      id="undefined-1"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value={1}
    />
  </span>
  <span
    className="MuiRating-decimal"
  >
    <label
      className="MuiRating-label"
      htmlFor="undefined-1-5"
      style={
        Object {
          "overflow": "hidden",
          "position": "absolute",
          "width": "0%",
          "zIndex": 1,
        }
      }
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
      <span
        className="MuiRating-visuallyhidden"
      >
        1.5 Stars
      </span>
    </label>
    <input
      checked={false}
      className="MuiRating-visuallyhidden"
      disabled={false}
      id="undefined-1-5"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value={1.5}
    />
    <label
      className="MuiRating-label"
      htmlFor="undefined-2"
      style={Object {}}
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
      <span
        className="MuiRating-visuallyhidden"
      >
        2 Stars
      </span>
    </label>
    <input
      checked={false}
      className="MuiRating-visuallyhidden"
      disabled={false}
      id="undefined-2"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value={2}
    />
  </span>
  <span
    className="MuiRating-decimal"
  >
    <label
      className="MuiRating-label"
      htmlFor="undefined-2-5"
      style={
        Object {
          "overflow": "hidden",
          "position": "absolute",
          "width": "0%",
          "zIndex": 1,
        }
      }
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
      <span
        className="MuiRating-visuallyhidden"
      >
        2.5 Stars
      </span>
    </label>
    <input
      checked={false}
      className="MuiRating-visuallyhidden"
      disabled={false}
      id="undefined-2-5"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value={2.5}
    />
    <label
      className="MuiRating-label"
      htmlFor="undefined-3"
      style={Object {}}
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
      <span
        className="MuiRating-visuallyhidden"
      >
        3 Stars
      </span>
    </label>
    <input
      checked={false}
      className="MuiRating-visuallyhidden"
      disabled={false}
      id="undefined-3"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value={3}
    />
  </span>
  <span
    className="MuiRating-decimal"
  >
    <label
      className="MuiRating-label"
      htmlFor="undefined-3-5"
      style={
        Object {
          "overflow": "hidden",
          "position": "absolute",
          "width": "0%",
          "zIndex": 1,
        }
      }
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
      <span
        className="MuiRating-visuallyhidden"
      >
        3.5 Stars
      </span>
    </label>
    <input
      checked={false}
      className="MuiRating-visuallyhidden"
      disabled={false}
      id="undefined-3-5"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value={3.5}
    />
    <label
      className="MuiRating-label"
      htmlFor="undefined-4"
      style={Object {}}
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
      <span
        className="MuiRating-visuallyhidden"
      >
        4 Stars
      </span>
    </label>
    <input
      checked={false}
      className="MuiRating-visuallyhidden"
      disabled={false}
      id="undefined-4"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value={4}
    />
  </span>
  <span
    className="MuiRating-decimal"
  >
    <label
      className="MuiRating-label"
      htmlFor="undefined-4-5"
      style={
        Object {
          "overflow": "hidden",
          "position": "absolute",
          "width": "0%",
          "zIndex": 1,
        }
      }
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
      <span
        className="MuiRating-visuallyhidden"
      >
        4.5 Stars
      </span>
    </label>
    <input
      checked={false}
      className="MuiRating-visuallyhidden"
      disabled={false}
      id="undefined-4-5"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value={4.5}
    />
    <label
      className="MuiRating-label"
      htmlFor="undefined-5"
      style={Object {}}
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
      <span
        className="MuiRating-visuallyhidden"
      >
        5 Stars
      </span>
    </label>
    <input
      checked={true}
      className="MuiRating-visuallyhidden"
      disabled={false}
      id="undefined-5"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      type="radio"
      value={5}
    />
  </span>
</span>
`;

exports[`StarRating renders with double star rating 1`] = `
<span
  aria-label="2.5 Stars"
  className="MuiRating-root MuiRating-readOnly"
  onMouseLeave={[Function]}
  onMouseMove={[Function]}
  role="img"
>
  <span
    className="MuiRating-decimal"
  >
    <span
      style={
        Object {
          "overflow": "hidden",
          "position": "absolute",
          "width": "0%",
          "zIndex": 1,
        }
      }
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
    <span
      style={Object {}}
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
  </span>
  <span
    className="MuiRating-decimal"
  >
    <span
      style={
        Object {
          "overflow": "hidden",
          "position": "absolute",
          "width": "0%",
          "zIndex": 1,
        }
      }
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
    <span
      style={Object {}}
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
  </span>
  <span
    className="MuiRating-decimal"
  >
    <span
      style={
        Object {
          "overflow": "hidden",
          "position": "absolute",
          "width": "50%",
          "zIndex": 1,
        }
      }
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
    <span
      style={Object {}}
    >
      <span
        className="MuiRating-icon MuiRating-iconEmpty"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
  </span>
  <span
    className="MuiRating-decimal"
  >
    <span
      style={
        Object {
          "overflow": "hidden",
          "position": "absolute",
          "width": "0%",
          "zIndex": 1,
        }
      }
    >
      <span
        className="MuiRating-icon MuiRating-iconEmpty"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
    <span
      style={Object {}}
    >
      <span
        className="MuiRating-icon MuiRating-iconEmpty"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
  </span>
  <span
    className="MuiRating-decimal"
  >
    <span
      style={
        Object {
          "overflow": "hidden",
          "position": "absolute",
          "width": "0%",
          "zIndex": 1,
        }
      }
    >
      <span
        className="MuiRating-icon MuiRating-iconEmpty"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
    <span
      style={Object {}}
    >
      <span
        className="MuiRating-icon MuiRating-iconEmpty"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
  </span>
</span>
`;

exports[`StarRating renders with minimal props 1`] = `
<span
  aria-label="5 Stars"
  className="MuiRating-root MuiRating-readOnly"
  onMouseLeave={[Function]}
  onMouseMove={[Function]}
  role="img"
>
  <span
    className="MuiRating-decimal"
  >
    <span
      style={
        Object {
          "overflow": "hidden",
          "position": "absolute",
          "width": "0%",
          "zIndex": 1,
        }
      }
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
    <span
      style={Object {}}
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
  </span>
  <span
    className="MuiRating-decimal"
  >
    <span
      style={
        Object {
          "overflow": "hidden",
          "position": "absolute",
          "width": "0%",
          "zIndex": 1,
        }
      }
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
    <span
      style={Object {}}
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
  </span>
  <span
    className="MuiRating-decimal"
  >
    <span
      style={
        Object {
          "overflow": "hidden",
          "position": "absolute",
          "width": "0%",
          "zIndex": 1,
        }
      }
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
    <span
      style={Object {}}
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
  </span>
  <span
    className="MuiRating-decimal"
  >
    <span
      style={
        Object {
          "overflow": "hidden",
          "position": "absolute",
          "width": "0%",
          "zIndex": 1,
        }
      }
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
    <span
      style={Object {}}
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
  </span>
  <span
    className="MuiRating-decimal"
  >
    <span
      style={
        Object {
          "overflow": "hidden",
          "position": "absolute",
          "width": "0%",
          "zIndex": 1,
        }
      }
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
    <span
      style={Object {}}
    >
      <span
        className="MuiRating-icon MuiRating-iconFilled"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </span>
    </span>
  </span>
</span>
`;
